File: two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        num_dict = dict()
        for i, num in enumerate(nums):
            comp = target - nums[i]
            if comp in num_dict:
                return [num_dict[comp], i]
            num_dict[num] = i
        return []


s = Solution()
print(s.twoSum([2, 7, 11, 15], 9))

File: Go_version/leetcode/twoSum.go
package leetcode

func TwoSum(nums []int, target int) []int {
	d := make(map[int]int)
	for i, num := range nums {
		comp := target - nums[i]
		if idx, ok := d[comp]; ok {
			return []int{idx, i}
		}
		d[num] = i
	}
	return []int{}
}
